### PR TITLE
Add performance counters

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Add performance counters debug tool

### CODE REVIEW

File: addons/common/functions/fnc_dumpPerformanceCounters.sqf [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+ * None
+ *
+ * Example:
+ * [] call zen_common_fnc_dumpPerformanceCounters
+ *
+ * Public: No
+ */
+#include "script_component.hpp"
+
+diag_log text "~~~ ZEN PERFORMANCE COUNTERS DUMP ~~~";
```
Comment (CreepPork) [inline addons/common/functions/fnc_dumpPerformanceCounters.sqf]: Couldn't you use the CBA macro `LOG`? And `TRACE_x` for all the others that have data?

File: addons/main/script_debug.hpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    #define COUNTER_NAME(x) GVAR(DOUBLES(x,counter))
+    #define CREATE_COUNTER(x) if (isNil "ZEN_COUNTERS") then {ZEN_COUNTERS = []}; COUNTER_NAME(x) = [QUOTE(COUNTER_NAME(x)), []]; ZEN_COUNTERS pushBack COUNTER_NAME(x)
+    #define BEGIN_COUNTER(x) if (isNil QUOTE(COUNTER_NAME(x))) then {CREATE_COUNTER(x)}; COUNTER_NAME(x) set [2, diag_tickTime]
+    #define END_COUNTER(x) (COUNTER_NAME(x) select 1) pushBack [COUNTER_NAME(x) select 2, diag_tickTime]
+#else
+    #define CREATE_COUNTER(x) /* disabled */
+    #define BEGIN_COUNTER(x) /* disabled */
+    #define END_COUNTER(x) /* disabled */
```
Comment (CreepPork) [inline addons/main/script_debug.hpp]: If disabled, then remove?

Comment (mharis001) [reply addons/main/script_debug.hpp]: What do you mean by remove? We still need the macros to be defined.

Comment (CreepPork) [reply addons/main/script_debug.hpp]: Whoops.

File: addons/common/functions/fnc_dumpPerformanceCounters.sqf (hunk 1 above)
Comment (mharis001) [reply addons/common/functions/fnc_dumpPerformanceCounters.sqf]: That would add the formatting that those macros come with. Output is cleaner like this.

Comment (CreepPork) [reply addons/common/functions/fnc_dumpPerformanceCounters.sqf]: But you have `diag_logs` couple of lines down that use formats. It's cleaner to use the macros.

Comment (CreepPork) [reply addons/common/functions/fnc_dumpPerformanceCounters.sqf]: Plus, is there any reason that `text` is called? Doesn't look very structured text to me.

Comment (veteran29) [reply addons/common/functions/fnc_dumpPerformanceCounters.sqf]: diag_log "asdf" => `"asdf"` in file
diag_log text "asdf" => `asdf` in file

Comment (CreepPork) [reply addons/common/functions/fnc_dumpPerformanceCounters.sqf]: Ah, I see.

**Pull request merged**